File: 0475-Heaters/0475-Heaters.cpp
class Solution {
public:
    int findRadius(vector<int>& houses, vector<int>& heaters) {        
        if (heaters.size() == 0) {
            return INT_MAX;
        }
        
        int radius = 0;
        sort(heaters.begin(), heaters.end());
        for (int h : houses) {
            auto right = lower_bound(heaters.begin(), heaters.end(), h);
            int rightRadius = (right == heaters.end()) ? h - heaters.back() : *right - h;
            int leftRadius = rightRadius;
            if (right != heaters.begin() && right != heaters.end()) {
                leftRadius = h - *(--right);
            }
            
            radius = max(radius, min(rightRadius, leftRadius));
        }
        
        return radius;
    }
};

File: 0401-0500/0475-Heaters/0475-Heaters.py
import bisect
class Solution:
    def findRadius(self, houses, heaters):
        """
        :type houses: List[int]
        :type heaters: List[int]
        :rtype: int
        """
        heaters.sort()
        radius = 0
        for house in houses:
            index = bisect.bisect_left(heaters, house)
            rightRadius = house - heaters[-1] if index == len(heaters) else heaters[index] - house
            leftRadius = house - heaters[index - 1] if index > 0 else rightRadius
            radius = max(radius, min(leftRadius, rightRadius))
        return radius
